feat: Update the congestion index calculation

luchengqi7 opened the pull request:
The congestion index based on the idea of excess travel time is now added (i.e., comparable to the TomTom congestion index). This index is preferred by VSP, and should replace the previous congestion index. 

To make it backward compatible, the previous congestion index is still kept, but deprecated. For the dashboard (SimWrapper), only the new excess travel time index is displayed. 

To avoid confusion, we name the new congestion index as "excess travel time index", in order to distinguish with the existing (deprecated) congestion index.

@rakow could you please help to check the dashboard part? Shall we replace the old plot or keep the old plot? And is there any quick test I can perform?

luchengqi7 committed:
Update LinkStats.java

luchengqi7 committed:
Update TrafficStatsCalculator.java

kainagel committed:
fix attribute reading

kainagel committed:
non-compiling suggestions written during joint meeting

luchengqi7 committed:
Merge branch 'main' into Update-congestion-index

luchengqi7 committed:
Merge branch 'main' into Update-congestion-index

luchengqi7 committed:
adding excess travel time index analysis

and update simwrapper dashboard

luchengqi7 committed:
Merge branch 'main' into Update-congestion-index

luchengqi7 committed:
Merge branch 'main' into Update-congestion-index

luchengqi7 committed:
Merge branch 'Update-congestion-index' of https://github.com/matsim-org/matsim-libs into Update-congestion-index

luchengqi7 committed:
small bug fixing

luchengqi7 committed:
minor updates

rewertvsp committed:
Merge branch 'main' into Update-congestion-index

rewertvsp committed:
Merge branch 'refs/heads/main' into Update-congestion-index

rewertvsp committed:
remove unused plot

rewertvsp committed:
adjust comments

kainagel commented:
Dear Chengqi, sorry that was dormant for so long.

I don't think that we need an additional specification for the averaging; the travel time index already does that.

A travel time index of 0.2 means that each trip in the average takes 20% longer than with the free speed travel time.  The average here is "over trips".

We now need to break this down into the pieces that we want.  The pieces are "hours" and "link types".  So a travel time index of 0.2 between 9am and 10am for "primary" would mean (in my understanding) that all vehicles that enter primary links between 9am and 10am are, in the average, 20% slower than with free speed.  I.e. you sum up the travel times of all vehicles on those links, and the free speed travel times of all vehicles on those links.  Only at the very end you divide. 

I.e. you do not take an average over links.  

Does this make sense?

rewertvsp committed:
Merge branch 'main' into Update-congestion-index

luchengqi7 commented:
Dear Prof. Nagel,

Thank you very much for the feedback! Yes, this sounds good. I will adapt that into the calculation, and update the PR. 

Best regards,
Chengqi Lu

luchengqi7 committed:
update traffic analysis

Finalize the excess travel time analysis, which replaces the current "congestion index".

luchengqi7 committed:
Merge branch 'main' into Update-congestion-index

luchengqi7 committed:
Merge branch 'main' into Update-congestion-index

luchengqi7 committed:
further clean up

luchengqi7 committed:
Merge branch 'Update-congestion-index' of https://github.com/matsim-org/matsim-libs into Update-congestion-index

luchengqi7 committed:
Merge branch 'main' into Update-congestion-index

rewertvsp committed:
Merge branch 'main' into Update-congestion-index
